autoscale-kubernetes-container/README.md

```diff
@@ -0,0 +1,222 @@
+# Autoscaler Kubernetes Module
+
+This repository consists of the source to build a pre-defined CAF approved
+container that includes the CAF autoscale application. Sample configuration is also supplied in the code base.
+
+To get the kubernetes autoscaler module up and running look at the [Quick Start](#quick-start) section below.
+
+### Installing the Autoscaler.
+The autoscaler can be installed to Kubernetes by using the kubectl command line tool.  
+ - `kubectl apply -f ./autoscaler.yaml`
+
+### Configuration
+
+Configuration of the AutoScaler is supported through the following environment variables:
+ - `CAF_RABBITMQ_MGMT_URL`  
+    Default: `http://rabbitmq:15672`  
+    Used to specify the RabbitMQ Management API Endpoint.  Alternatively `CAF_RABBITMQ_HOST` and `CAF_RABBITMQ_MGMT_PORT` may instead be specified individually.
+
+ - `CAF_RABBITMQ_MGMT_USERNAME`  
+    Default: `guest`  
+    Used to specify the username used to connect to RabbitMQ.  If `CAF_RABBITMQ_MGMT_USERNAME` is not specified then `CAF_RABBITMQ_USERNAME` will also be checked before falling back to the default.
+
+ - `CAF_RABBITMQ_MGMT_PASSWORD`  
+    Default: `guest`  
+    Used to specify the password used to connect to RabbitMQ.  If `CAF_RABBITMQ_MGMT_PASSWORD` is not specified then `CAF_RABBITMQ_PASSWORD` will also be checked before falling back to the default.  
+
+ - `CAF_AUTOSCALER_RABBITMQ_MEMORY_QUERY_FREQ`  
+    Default: `10`  
+    Number of whole seconds that the service should wait between sending RabbitMQ memory status requests.
+
+ - `CAF_AUTOSCALER_SCALING_DELAY`  
+    Minimum: `1`  
+    Default: `10`  
+    Used to specify the number of queue statistics records that must be acquired sequentially before performing analysis upon them and issuing a scaling recommendation. This means the period in time between making scaling recommendations is the service interval multiplied by the scalingDelay. This number should be small enough to respond to varying load but big enough to avoid erratic behaviour.
+
+ - `CAF_AUTOSCALER_BACKLOG_GOAL`  
+    Default: `300`  
+    Used to specify the amount of time, in seconds, that we ideally want to complete the current backlog of messages in. This is effectively a quality of service parameter, where lower will trigger more aggressive scaling.
+
+ - `CAF_AUTOSCALER_MAXIMUM_INSTANCES`  
+    Default: `100`  
+    Used to specify the maximum number of instances that any worker can be scaled to.
+
+ - `CAF_AUTOSCALER_METRIC`  
+    Default: `rabbitmq`  
+
+ - `CAF_AUTOSCALER_KUBERNETES_NAMESPACES`
+    Default: `default`
+    Used to specify the Kubernetes namespaces, comma separated, to search for deployments.
+ 
+ - `CAF_AUTOSCALER_RESOURCE_ID_SEPARATOR`
+    Default: `:`
+    Description: The separator used when storing namespace/deployment source configuration. 
```

Comment:
Does this need to be configurable?  Is this in case someone uses a `:` in their namespace or service?  Is that actually allowed?

Reply:
Removed